src/generic_ui/polymer/logs.ts

```diff
@@ -4,26 +4,63 @@ import translator_module = require('../scripts/translator');
 declare var PolymerExpressions: any;
 PolymerExpressions.prototype.$$ = translator_module.i18n_t;
 
-declare var bringUproxyToFront :() => void;
-declare var getLogs :() => Promise<string>;
+const extensionId = 'pjpcdnccaekokkkeheolmpkfifcbibnj';
 
 Polymer({
   logs: '',
   loadingLogs: true,
   openUproxy: function() {
-    // TODO: add a pop-out icon that calls this function.
-    bringUproxyToFront();
+    if (window.chrome) {
+      chrome.runtime.sendMessage(extensionId, {
+        openWindow: true
+      });
+    } else {
+      window.postMessage({
+        showPanel: true,
+        data: false
+      }, '*');
+    }
   },
   created: function() {
     // Default language to English.
     var language = window.location.href.split('lang=')[1] || 'en';
-    translator_module.i18n_setLng(language.substring(0,2));
+    translator_module.i18n_setLng(language.substring(0, 2));
   },
   ready: function() {
     // Expose global ui object in this context.
-    getLogs().then((logs) => {
+
+    let fulfill: Function;
+    let reject: Function;
```

Comment:
I'd go for more descriptive variable names than `fulfill` and `reject here.  On the other hand, it might be easier to just have another method that takes the loaded logs and call that directly instead of creating the promise at all (e.g. `chrome.runtime.sendMessage(..., (reply) => { this.populateLogs(reply.logs); });`)